src/fairseq2/checkpoint.py

```diff
@@ -456,7 +466,10 @@ def save_consolidated_model(self, step_nr: int, model: Module) -> None:
     def load_model(
         self, step_nr: int, out: Module, *, device: Optional[Device] = None
     ) -> None:
-        model_file = self._checkpoint_dir.joinpath(f"step_{step_nr}/model.pt")
+        model_file = self.get_model_checkpoint_path(step_nr=step_nr)
+        if model_file is None:
+            # This should never happen, because if the step_nr is provided, model_file is always non-empty.
```

Comment:
This is not entirely correct. For FSDP checkpoints for instance, a checkpoint might have no consolidated model. So I would omit this comment and change the exception message to "Training step {step_number} has no saved model." similar to line 478.